app/src/sharedTest/java/org/oppia/app/player/content/UrlImageParserTest.kt

```diff
@@ -0,0 +1,75 @@
+package org.oppia.app.player.content
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.player.exploration.ExplorationActivity
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Singleton
+
+/** Tests for [UrlImageParser]. */
+@RunWith(AndroidJUnit4::class)
+class UrlImageParserTest {
+
+  @Test
+  fun testUrlImageParser_loadHtmlContent_isDisplayed() {
+    ActivityScenario.launch(ExplorationActivity::class.java).use {
+      onView(withId(org.oppia.app.R.id.recyclerview)).check(matches(isDisplayed()))
+      pauseTestFor(3000) // pause as image loading takes 2 to 3 secs.
```

Comment:
We shouldn't put artificial delays in tests. They can cause tests to become flaky and fail arbitrarily since it's not actually clear how long is a good time to pause for. Also, they can cause tests to run much longer than is necessary since they're always sleeping for the same amount of time.

In this case, the issue that needs to be addressed is using a network in a test. We should do this by creating an interface to load images that wraps Glide, and replace it in tests with something that returns a simple fake drawable. See https://stackoverflow.com/a/46286760 for a similar explanation, though you'll need to "Daggerify" the approach. Please let me know if you'd like to discuss this further.

Reply:
@BenHenning I have used interface to load image. But I am not able to write testcase for it.

Reply:
Duplicate of this PR is created [#205](https://github.com/oppia/oppia-android/pull/205)